agent/util-scripts/pbench-register-tool-set

```diff
@@ -129,83 +129,12 @@ if [[ -z ${the_tool_set} ]]; then
 	exit 1
 fi
 
-
-# This function checks a --remotes file argument  and if everything
-# goes OK it returns 0 (it also returns 0 if the --remotes argument is
-# not a file).  Otherwise, it logs an error and returns 1.
-
-# This is an emasculated version of the code that pbench-register-tool
-# uses to do this checking: in addition to the checking, that code
-# gathers information from the file into a couple of arrays. Since we
-# don't need those for pbench-register-tool-set the data gathering is
-# elided in this function.
-
-function check_remotes_file () {
-	local remotes_arg="${1}"
-	local lineno
-
-	if [[ ${remotes_arg::1} == "@" ]]; then
-		# We are dealing with a file containing the list of remotes.
-
-		if [[ -n ${labels_arg} ]]; then
-			error_log "--labels=${labels_arg} not allowed with remotes file (${remotes_arg})"
-			return 1
-		fi
-		remotes_file=${remotes_arg#*@}
-		if [[ ! -e ${remotes_file} ]]; then
-			error_log "--remotes=@${remotes_file} specifies a file that does not exist"
-			return 1
-		fi
-
-		lineno=0
-		while read line; do
-			(( lineno++ ))
-			if [[ ${line::1} == "#" ]]; then
-				# Ignore comments
-				(( lineno++ ))
-				continue
-			fi
-			IFS=',' read -ra parts <<< "${line}"
-			case "${#parts[@]}" in
-				0|1|2)
-				;;
-				*)
-				error_log "--remotes=@${remotes_file} contains an invalid file format, expected lines with \"<hostname>[,<label>]\" at line #${lineno}"
-				return 1
-				;;
-			esac
-		done < ${remotes_file}
-	fi
-
-	return 0
-}
-
-# call the function to check the argument
-check_remotes_file "${remotes_arg}"
-ret="${?}"
-if [[ "${ret}" != 0 ]]; then
+pbench-register-tool --name=vmstat --test-args ${remotes_arg}${labels_arg}
```

Comment:
Specifying what amounts to a dummy `--name` argument here seems kind of hokey.  Can you move the validation of the `--name` switch in `agent/util-scripts/pbench-register-tool` to after the validation of the remotes arguments (i.e., after the `if [[ ${_do_test_args} -ne 0 ]]` test at line 325)?  That would allow this code to simply omit the `--name` switch.

Reply:
OTOH, it would be  nice to see a missing `--name` flagged early, before the more arcane errors. I changed it to allow a missing `--name` iff `--test-args` is specified.